spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/elasticsearch/ElasticsearchRestHealthIndicator.java

```diff
@@ -16,75 +16,31 @@
 
 package org.springframework.boot.actuate.elasticsearch;
 
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
-import java.util.Map;
-
-import org.apache.http.HttpStatus;
-import org.apache.http.StatusLine;
-import org.elasticsearch.client.Request;
-import org.elasticsearch.client.Response;
 import org.elasticsearch.client.RestClient;
 import org.elasticsearch.client.RestHighLevelClient;
 
-import org.springframework.boot.actuate.health.AbstractHealthIndicator;
-import org.springframework.boot.actuate.health.Health;
 import org.springframework.boot.actuate.health.HealthIndicator;
-import org.springframework.boot.json.JsonParser;
-import org.springframework.boot.json.JsonParserFactory;
-import org.springframework.util.StreamUtils;
 
 /**
- * {@link HealthIndicator} for an Elasticsearch cluster using a {@link RestClient}.
+ * {@link HealthIndicator} for an Elasticsearch cluster using a {@link RestClient} or a
+ * {@link RestHighLevelClient}.
  *
  * @author Artsiom Yudovin
  * @author Brian Clozel
  * @author Filip Hrisafov
  * @since 2.1.1
+ * @deprecated since 2.6.0 for removal in 2.8.0 in favor of
+ * {@link ElasticsearchRestClientHealthIndicator}
  */
-public class ElasticsearchRestHealthIndicator extends AbstractHealthIndicator {
-
-	private static final String RED_STATUS = "red";
-
-	private final RestClient client;
-
-	private final JsonParser jsonParser;
+@Deprecated
+public class ElasticsearchRestHealthIndicator extends ElasticsearchRestClientHealthIndicator {
```

Comment:
I really wanted to avoid deprecating this. However, it is not possible to use this class when the `RestHighLevelClient` is not available.